backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ProductStockSammMapper.java

```diff
@@ -107,69 +100,46 @@ public ProductStockSammDto toSamm(Stock stock) {
-                        pos -> pos.getAllocatedStocks().stream().
-                                mapToDouble(stock ->
-                                        stock.getQuantityOnAllocatedStock().getQuantityNumber()
-                                ).sum()
-                ).sum();
-
-        // This is just a quickfix in line with the above-mentioned restriction that
-        // we are currently supporting only an aggregated stock. It is assumed that all
-        // stocks are using the same unit. If, for example, one stock uses kilograms and
-        // another stock uses metric tonnes for the same material, this will of course
-        // lead to faulty data.
-        String measurementUnitString = samm.getPositions().stream().findFirst()
-            .stream().findFirst().get().getAllocatedStocks().stream().findFirst()
-            .get().getQuantityOnAllocatedStock().getMeasurementUnit();
-        measurementUnitString = measurementUnitString.replace("unit:", "");
-        MeasurementUnit measurementUnit = MeasurementUnit.valueOf(measurementUnitString);
-
-        // Another quickfix. This will retrieve the lastUpdatedOn timestamp from
-        // the first position found and apply it to the accumulated stock.
-        Date lastUpdatedOnDate = samm.getPositions().stream().findFirst()
-            .get().getLastUpdatedOnDateTime();
-
-
-        // determine material
-
-        Material foundMaterial = materialService.findByOwnMaterialNumber(samm.getMaterialNumberCustomer());
-        MaterialPartnerRelation materialPartnerRelation = mprService.find(foundMaterial, partner);
-        MaterialDto foundMaterialDto = new MaterialDto(foundMaterial.isMaterialFlag(), foundMaterial.isProductFlag(),
-            foundMaterial.getOwnMaterialNumber(), materialPartnerRelation.getPartnerMaterialNumber(), foundMaterial.getMaterialNumberCx(),
-            foundMaterial.getName());
-
-        // find bpns - we use the first one as we currently only have one site per partner.
-        // alternative would be to inject the bpnl of the partner.
-        // also we set the bpn here always to the site
-        String atSiteBpns =
-                samm.getPositions()
-                        .stream().findFirst().get().getAllocatedStocks()
-                        .stream().findFirst().get().getSupplierStockLocationId().getLocationId();
-
-        PartnerDto supplierPartner = modelMapper.map(partner, PartnerDto.class);
-
-        return new PartnerProductStockDto(
-                foundMaterialDto,
-                quantity,
-                measurementUnit,
-                atSiteBpns,
-                supplierPartner,
-                lastUpdatedOnDate
-        );
+    public List<PartnerProductStock> sammToPartnerProductStocks(ProductStockSammDto samm, Partner partner) {
+        ArrayList<PartnerProductStock> output = new ArrayList<>();
+        Material material = materialService.findByOwnMaterialNumber(samm.getMaterialNumberCustomer());
```

Comment:
This logic is not appropriate. If Catena-X number is given, then this number must be used. It's like a cascade (use if given):
- catenaX
- supplier
- customer

Customer number is mandatory, rest is optional :)